src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -0,0 +1,465 @@
+import net.sf.jabref.logic.importer.fileformat.mods.LocationDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.ModsCollectionDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.ModsDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.NameDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.NamePartDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.NoteDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.OriginInfoDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.PartDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.PhysicalLocationDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.PlaceDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.PlaceTermDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.RecordInfoDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.RelatedItemDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.StringPlusLanguage;
+import net.sf.jabref.logic.importer.fileformat.mods.StringPlusLanguagePlusAuthority;
+import net.sf.jabref.logic.importer.fileformat.mods.StringPlusLanguagePlusSupplied;
+import net.sf.jabref.logic.importer.fileformat.mods.SubjectDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.TitleInfoDefinition;
+import net.sf.jabref.logic.importer.fileformat.mods.UrlDefinition;
+import net.sf.jabref.logic.util.FileExtensions;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import com.google.common.base.Joiner;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ *
+ * Importer for the MODS format.<br>
+ * More details about the format can be found here <a href="http://www.loc.gov/standards/mods/">http://www.loc.gov/standards/mods/</a>. <br>
+ * The newest xml schema can also be found here <a href="www.loc.gov/standards/mods/mods-schemas.html.">www.loc.gov/standards/mods/mods-schemas.html.</a>.
+ *
+ */
+public class ModsImporter extends ImportFormat {
+
+    private static final Log LOGGER = LogFactory.getLog(ModsImporter.class);
+    private static final String KEYWORD_SEPARATOR = JabRefPreferences.getInstance().getImportFormatPreferences()
+            .getKeywordSeparator();
+
+    private static final Pattern MODS_PATTERN = Pattern.compile("<mods .*>");
+
+
+    @Override
+    protected boolean isRecognizedFormat(BufferedReader input) throws IOException {
+        return input.lines().anyMatch(line -> MODS_PATTERN.matcher(line).find());
+    }
+
+    @Override
+    public ParserResult importDatabase(BufferedReader input) throws IOException {
+        Objects.requireNonNull(input);
+
+        List<BibEntry> bibItems = new ArrayList<>();
+
+        JAXBContext context;
+        try {
+            context = JAXBContext.newInstance("net.sf.jabref.logic.importer.fileformat.mods");
```

Comment:
Just a non-so-critical remark: `JAXBContext` is known to be an object that is costly to construct and doing this often might result in performance issues. Hence, you typically build it once and keep it. Could you extract `context` as an instance variable? Then, check on execution of `importDatabase` if it is null and if so initialize it.